glyphs-reader/src/font.rs

```diff
@@ -1110,6 +1140,52 @@ fn raw_feature_to_feature(feature: RawFeature) -> Result<FeatureSnippet, Error>
     Ok(FeatureSnippet(code))
 }
 
+fn copy_metric_to_masters(
+    font: &mut RawFont,
+    name: &str,
+    set_fn: impl Fn(&mut FontMaster, OrderedFloat<f64>),
+) {
+    let Some(metrics) = &font.metrics else {
+        return;
+    };
+    let Some(idx) = metrics.iter().position(|metric| metric.type_.as_ref().map(|n| n.as_str() == name).unwrap_or_default()) else {
+        return;
+    };
+    for master in font.font_master.iter_mut() {
+        let Some(values) = &master.metric_values else {
+            continue;
+        };
+        let Some(metric) = values.get(idx) else {
+            continue;
+        };
+        let Some(value) = metric.pos else {
+            continue;
+        };
+        set_fn(master, value);
+    }
+}
+
+fn reorg_metric_values(font: &mut RawFont) {
+    copy_metric_to_masters(font, "ascender", |master, v| {
+        master.ascender = Some(v);
+    });
+    copy_metric_to_masters(font, "descender", |master, v| {
+        master.descender = Some(v);
+    });
+    copy_metric_to_masters(font, "baseline", |master, v| {
+        master.baseline = Some(v);
+    });
+    copy_metric_to_masters(font, "cap height", |master, v| {
+        master.cap_height = Some(v);
+    });
+    copy_metric_to_masters(font, "x-height", |master, v| {
+        master.x_height = Some(v);
+    });
+    for master in font.font_master.iter_mut() {
+        master.metric_values = None;
```

Comment:
I understand the convenience of doing `font.default_master().ascender` etc. But maybe instead of copying the values, you could add some accessors (e.g. `ascender()`) that would look up the respective metric into the metrics_values?

Reply:
I need the attribute to parse the objects nicely but if I use RawFontMaster and turn it into FontMaster there's a clear place to shuffle the attributes into metric_values so ... probably I should just not be lazy and actually do that.

Reply:
Done. I did take the liberty of storing a map<name, value> instead of having a parent with names and a child with a parallel value structure because that's just annoying and especially annoying in Rust, you end up with an Rc and a messy lookup.